pkg/sources/udsource/utils/offset.go

```diff
@@ -17,21 +17,25 @@ limitations under the License.
 package utils
 
 import (
+	"encoding/base64"
+
 	sourcepb "github.com/numaproj/numaflow-go/pkg/apis/proto/source/v1"
 
 	"github.com/numaproj/numaflow/pkg/isb"
 )
 
 // simpleSourceOffset is a simple implementation of isb.Offset from the source side.
 type simpleSourceOffset struct {
+	// NOTE: offset is base64 encoded string because we use offset to construct message ID
+	// and message ID is a string.
 	offset       string
 	partitionIdx int32
 }
 
-func NewSimpleSourceOffset(o string, p int32) isb.Offset {
+func NewSimpleSourceOffset(offset *sourcepb.Offset) isb.Offset {
 	return &simpleSourceOffset{
-		offset:       o,
-		partitionIdx: p,
+		offset:       base64.StdEncoding.EncodeToString(offset.GetOffset()),
+		partitionIdx: offset.GetPartitionId(),
 	}
 }
 
```

Comment:
Should we include partitionIdx in the `String()` of offset as you did in your first commit of the PR? Since partitionIdx is now an integral part of simpleSourceOffset.